src/main/java/formflow/library/ValidationService.java

```diff
@@ -158,4 +163,25 @@ private Map<String, List<String>> performFieldLevelValidation(String flowName, F
 
     return validationMessages;
   }
+
+  public static Map<String, Boolean> getRequiredInputs(String flowName) {
+    if (requiredInputs.isEmpty()) {
+      Class<?> flowClass;
+
+      try {
+        flowClass = Class.forName(inputConfigPath + StringUtils.capitalize(flowName));
+      } catch (ReflectiveOperationException e) {
+        throw new RuntimeException(e);
+      }
+
+      Field[] declaredFields = flowClass.getDeclaredFields();
+      for (Field field : declaredFields) {
+        if (Arrays.stream(field.getAnnotations())
+            .anyMatch(annotation -> requiredAnnotationsList.contains(annotation.annotationType().getName()))) {
+          requiredInputs.put(field.getName(), true);
```

Comment:
Why is this a map and not just a list?  Seems like everything mapping to 'true' makes the mapping unnecessary?

Reply:
Was trying to optimize for instant look up in a map vs O(n) if an application has a lot of inputs. Does it feel like over optimizing?

Reply:
That makes sense.